Sources/Amplitude/Amplitude.m

```diff
@@ -96,6 +96,7 @@ @interface Amplitude ()
 @property (nonatomic, copy, readwrite, nullable) NSString *userId;
 @property (nonatomic, copy, readwrite) NSString *deviceId;
 @property (nonatomic, copy, readwrite) NSString *contentTypeHeader;
+@property (nonatomic, assign) BOOL updatingCurrently;
```

Comment:
We only store events in uploadEvents. If we don't expose this property to the test case, we cannot test if all events are saved in the storage. If we have multiple flushQueue in one test case, only the first one will trigger the storeEvents, and others will stop because of https://github.com/amplitude/Amplitude-iOS/blob/main/Sources/Amplitude/Amplitude.m#L733

Reply:
But I'm open to discussing if we should do this change.

Reply:
This is not test driven, it's complicating the logic. 

You can change/separate the test cases. But adding unnecessary variable is not a good idea.